### PR TITLE
[Python] add `float` to reserved word list

### PR DESCRIPTION
**Pull request by wing328**
- add `float` to reserved word list
- update tests

### PR checklist
 
- [ ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [ ] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [ ] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.1.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [ ] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by wing328** [comment]
@multani  I've filed this change to fix the `float` property issue

**Comment by multani** [comment]
> @multani I've filed this change to fix the `float` property issue. Please review when you've time.

Ah, very nice! :+1: 

Note that generally speaking, it's bit more complicated: if you have a field named after anything on the right hand side of the typing information (like `Annotated`, `Field`, `List`, or anything like this), this would fail the same way.

I haven't checked all the generator, so perhaps it lower case all the field names and it's less likely to create a problem.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractPythonCodegen.java [hunk 1]
```diff
@@ -110,6 +110,9 @@ public AbstractPythonCodegen() {
                         "header_params", "form_params", "local_var_files", "body_params", "auth_settings",
                         // @property
                         "property",
+                        // pydantic keyword
```
Comment (multani) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractPythonCodegen.java:113]: `schema` may be Pydantic-specific, but not the others I think:

```suggestion
                        // typing keywords
```

Note that we may have the same problem with any basic types (int, bool, datetime, etc.)

**Review by multani — CHANGES_REQUESTED** [review]
Looks good to me, except for the caveats / limitations I mentioned. I don't think they are new though :)

**Review by multani — CHANGES_REQUESTED** [review]
@wing328 Can you also uncomment https://github.com/OpenAPITools/openapi-generator/blob/3fcf5584c6124d38dbf88b349cde697b2a0b5f5f/modules/openapi-generator/src/test/resources/3_0/python/petstore-with-fake-endpoints-models-for-testing.yaml#L1652-L1657?

You fix should address this TODO as well :+1:

**Review by multani — APPROVED** [review]

**Pull request merged**